### ISSUE TITLE
Improve usability for search/replace "with confirm"

### ISSUE THREAD

**Issue by kbilsted** [neovim/neovim#31667]
### Problem

when doing search/replace the user is presented with a "cryptic" help in the form of 
` (y/n/a/q/l/^E/^Y)`

### Expected behavior

I propose we elaborate a bit on the help text to cater for the inexperienced by explaining each option, for example 
` (y)es/(n)o/(a)ll/(q) stop/(l) replace and stop/(^E) scroll down/(^Y) scroll up` 

or if this is too verbose we could skip the readily understandable ones (y/n/a/q)
` (y/n/a/q/l (replace+stop)/^E (down)/^Y (up)) `

**Comment by zeertzjq** [comment]
> or if this is too verbose we could skip the readily understandable ones (y/n/a/q)

The easily understandable ones are the most useful ones. The remaining ones aren't that useful.

> when doing search/replace the user is presented with a "cryptic" help in the form of

Is that really what a newcomer will encounter? This prompt won't be show unless the user explicitly add the `c` flag.

**Comment by kbilsted** [comment]
@zeertzjq  I agree with all your points. However, I'm not sure I understand what you are trying to say? 

You think it is helpful that the available options are displayed, but it is not helpful that the options are explained?

**Comment by justinmk** [comment]
How about this:

    y/n/(a)ll/(q)uit/(l)ast/(^E) scroll up/(^Y) scroll down

**Comment by kbilsted** [comment]
I think `last` doesn't explain very well what it does

**Comment by justinmk** [comment]
that is how it's documented and that is what it stands for. the dialog isn't meant to fully explain everything, only to give useful hints

**Comment by siddhantdev** [comment]
Can I work on this?

**Comment by luukvbaal** [comment]
> Can I work on this?

You're welcome to! Note that any change to this message will conflict with #31525. But it should be no problem to rebase one on top of the other.

**Comment by siddhantdev** [comment]
I looked into it. Seems to me like make the change just requires changing the call to `smsg` in `nvim/ex_cmds.c` at line 3806. That hasn't been touched in https://github.com/neovim/neovim/pull/31525 at all. So it should be fine.

**Comment by luukvbaal** [comment]
> I looked into it. Seems to me like make the change just requires changing the call to smsg in nvim/ex_cmds.c at line 3806.

Correct.

> That hasn't been touched in https://github.com/neovim/neovim/pull/31525 at all.

While it doesn't change the message, it definitely changes the code around it and in fact completely replaces the `smsg` call: https://github.com/neovim/neovim/pull/31525/files#diff-79c00103dd452f1d0d57ea0e308f821d6a85d139f2327098e161ff2869bdc52bL3808. (Still not a problem, whichever change is merged first, but it does conflict and would require rebasing...)

**Comment by siddhantdev** [comment]
Perhaps it would be easier to just wait for https://github.com/neovim/neovim/pull/31525 to get merged or make the change in that commit? This is a small change with low priority anyway. Seems like the easier option.

**Comment by luukvbaal** [comment]
I opened #31787 (so we can discuss the change while #31525 is yet unmerged, if needed).
